### PR TITLE
Initial version of handling multiple WireMock instances, add common test methods

### PR DESCRIPTION
**Pull request by tMaxx**
And KnotxTestUtils gets deprecated in favor of more modular approach.

### CODE REVIEW

File: src/main/java/io/knotx/junit5/util/RequestUtil.java [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+   * @return reactivex wrapper
+   */
+  static Observable<HttpClientResponse> asyncRequest(
+      HttpClient client,
+      HttpMethod method,
+      int port,
+      String domain,
+      String uri,
+      java.util.function.Consumer<HttpClientRequest> requestBuilder) {
+    return Observable.unsafeCreate(
```
Comment (tomaszmichalak) [inline src/main/java/io/knotx/junit5/util/RequestUtil.java]: You can wrap HttpClient with WebClient and do all this stuff RX-way.

File: src/main/java/io/knotx/junit5/util/RequestUtil.java [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+
+  /**
+   * Safely execute onSuccess handler on given result, passing checks to given test context
+   *
+   * @param context test context
+   * @param result to which subscribe
+   * @param onSuccess result handler, can throw exceptions
+   */
+  static <T> void subscribeToResult_shouldSucceed(
+      VertxTestContext context, Single<T> result, Consumer<T> onSuccess) {
```
Comment (tomaszmichalak) [inline src/main/java/io/knotx/junit5/util/RequestUtil.java]: onSuccess -> assertions

File: src/main/java/io/knotx/junit5/util/RequestUtil.java (hunk 1 above)
Comment (tMaxx) [reply src/main/java/io/knotx/junit5/util/RequestUtil.java]: Due to its limited use (one occurrence in knotx-core) I think we can safely mark this as deprecated and throw it out of the window on next opportunity.

I'll move it back to KnotxTestUtils where it belongs, it was added to RequestUtil under assumption that it was more widely used.

File: src/main/java/io/knotx/junit5/wiremock/KnotxWiremockExtension.java [hunk 3]
```diff
@@ -143,16 +154,31 @@ private WireMockServer setupWiremockServer(KnotxWiremock knotxWiremock, Store st
...
     port = server.port();
-    WireMock.configureFor("localhost", port);
+    getOrCreateWiremock(port);
     map.put(port, server);
 
     return server;
   }
 
+  private static WireMock getOrCreateWiremock(int port) {
+    if (wiremockMap.containsKey(port)) {
```
Comment (tomaszmichalak) [inline src/main/java/io/knotx/junit5/wiremock/KnotxWiremockExtension.java]: I think that this if statement should be after `wiremockMapLock.lock()`,

File: src/main/java/io/knotx/junit5/KnotxTestUtils.java [hunk 4]
```diff
@@ -15,16 +15,16 @@
...
 import io.reactivex.Observable;
 import io.vertx.core.http.HttpMethod;
 import io.vertx.reactivex.core.http.HttpClient;
 import io.vertx.reactivex.core.http.HttpClientRequest;
 import io.vertx.reactivex.core.http.HttpClientResponse;
 import java.io.IOException;
-import java.nio.charset.StandardCharsets;
 import java.util.function.Consumer;
 
+@Deprecated
```
Comment (Skejven) [inline src/main/java/io/knotx/junit5/KnotxTestUtils.java]: Let's either remove it or add note to use `io.knotx.junit5.util.FileReader` instead.

File: src/main/java/io/knotx/junit5/util/RequestUtil.java [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.knotx.junit5.util;
+
+import io.reactivex.Single;
+import io.reactivex.functions.Consumer;
+import io.vertx.junit5.VertxTestContext;
+
+public interface RequestUtil {
```
Comment (Skejven) [inline src/main/java/io/knotx/junit5/util/RequestUtil.java]: Using java `interface` to make util class is not a good practice. There is wide discussion about that you may find in the Internet. Please change it to `final class`.

File: src/main/java/io/knotx/junit5/wiremock/KnotxWiremockExtension.java (hunk 3 above)
Comment (tMaxx) [reply src/main/java/io/knotx/junit5/wiremock/KnotxWiremockExtension.java]: done

File: src/main/java/io/knotx/junit5/KnotxTestUtils.java (hunk 4 above)
Comment (tMaxx) [reply src/main/java/io/knotx/junit5/KnotxTestUtils.java]: Note added to method

File: src/main/java/io/knotx/junit5/util/RequestUtil.java [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.knotx.junit5.util;
+
+import io.reactivex.Single;
+import io.reactivex.functions.Consumer;
+import io.vertx.junit5.VertxTestContext;
+
+public final class RequestUtil {
```
Comment (Skejven) [inline src/main/java/io/knotx/junit5/util/RequestUtil.java]: Now empty private constructor :)

File: src/main/java/io/knotx/junit5/wiremock/KnotxWiremockExtension.java [hunk 7]
```diff
@@ -162,17 +213,29 @@ private void cleanupWiremock(ExtensionContext context) {
...
+  private WiremockServerMap getWiremockServerMap(Store store) {
     return store.getOrComputeIfAbsent(
-        WIREMOCK_INSTANCES_MAP_STORE_KEY, o -> new WiremockMap(), WiremockMap.class);
+        WIREMOCK_SERVER_INSTANCES_MAP_STORE_KEY,
+        o -> new WiremockServerMap(),
+        WiremockServerMap.class);
   }
 
-  private class WiremockMap extends HashMap<Integer, WireMockServer> {}
+  private static class WiremockMap extends HashMap<Integer, WireMock> {}
```
Comment (Skejven) [inline src/main/java/io/knotx/junit5/wiremock/KnotxWiremockExtension.java]: Wouldn't be simpler to just have `wiremockMap`  defined as `HashMap<Integer, WireMock>`?

File: src/main/java/io/knotx/junit5/util/RequestUtil.java (hunk 6 above)
Comment (tMaxx) [reply src/main/java/io/knotx/junit5/util/RequestUtil.java]: done

**Pull request merged**